### PR TITLE
Fix 'added, removed, shared members list were wrong when updating rep…

### PR DESCRIPTION
**Pull request by vanlinha3k35pbc**

### CODE REVIEW

**Comment by vanlinha3k35pbc** [comment]
Fix Bug #33177 http://dev.framgia.com/redmine/issues/33177

File: app/workers/edit_repeat_worker.rb [hunk 1]
```diff
@@ -17,12 +24,42 @@ def perform added_member_ids, removed_member_ids, schedule_id, announce
 
       if schedule.valid?
         schedule.save
-        repeated_schedule_ids << schedule.id
+
+        if sample_schedule_id != schedule_id
```
Comment (dieunb) [inline app/workers/edit_repeat_worker.rb]: unless

File: app/workers/edit_repeat_worker.rb [hunk 2]
```diff
@@ -17,12 +24,42 @@ def perform added_member_ids, removed_member_ids, schedule_id, announce
...
+          added_member_ids = new_member_ids - old_member_ids
+          removed_member_ids = old_member_ids - new_member_ids
+          shared_member_ids = new_member_ids & old_member_ids
+        else
+          added_member_ids = sample_added_member_ids
+          removed_member_ids = sample_removed_member_ids
+          shared_member_ids = sample_shared_member_ids
+        end
+
+        added_member_ids.each do |member_id|
```
Comment (dieunb) [inline app/workers/edit_repeat_worker.rb]: one line code

File: app/workers/edit_repeat_worker.rb [hunk 3]
```diff
@@ -17,12 +24,42 @@ def perform added_member_ids, removed_member_ids, schedule_id, announce
...
+          added_list[member_id] << schedule_id
+        end
+
+        removed_member_ids.each do |member_id|
```
Comment (dieunb) [inline app/workers/edit_repeat_worker.rb]: one line code

File: app/workers/edit_repeat_worker.rb [hunk 4]
```diff
@@ -17,12 +24,42 @@ def perform added_member_ids, removed_member_ids, schedule_id, announce
...
+          removed_list[member_id] << schedule_id
+        end
+
+        shared_member_ids.each do |member_id|
```
Comment (dieunb) [inline app/workers/edit_repeat_worker.rb]: one line code

File: app/workers/edit_repeat_worker.rb [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
-    shared_member_ids = schedule_sample.member_ids - added_member_ids
+    sample_shared_member_ids = schedule_sample.member_ids - sample_added_member_ids
+
+    added_list = Hash.new {|hash, key| hash[key] = Array.new}
+    removed_list = Hash.new {|hash, key| hash[key] = Array.new}
+    shared_list = Hash.new {|hash, key| hash[key] = Array.new}
+
     schedules.each do |schedule|
+      schedule_id = schedule.id
+      old_member_ids = schedule.member_ids if sample_schedule_id != schedule_id
```
Comment (dieunb) [inline app/workers/edit_repeat_worker.rb]: unless

File: app/workers/edit_repeat_worker.rb [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
-    schedule_sample = Schedule.find schedule_id
+  def perform sample_schedule_id, announce, sample_added_member_ids, sample_removed_member_ids
+    schedule_sample = Schedule.find sample_schedule_id
     repeat = Repeat.find schedule_sample.repeat_id
     schedules = repeat.schedules
-    repeated_schedule_ids = Array.new
-    shared_member_ids = schedule_sample.member_ids - added_member_ids
+    sample_shared_member_ids = schedule_sample.member_ids - sample_added_member_ids
+
+    added_list = Hash.new {|hash, key| hash[key] = Array.new}
```
Comment (dieunb) [inline app/workers/edit_repeat_worker.rb]: Viết 3 dòng gọn lại được không?

**Pull request merged**